salt/states/http.py

```diff
@@ -94,28 +107,40 @@ def query(name, match=None, match_type='string', status=None, wait_for=None, **k
 
     if match is not None:
         if match_type == 'string':
-            if match in data.get('text', ''):
+            if str(match) in data.get('text', ''):
                 ret['result'] = True
                 ret['comment'] += ' Match text "{0}" was found.'.format(match)
             else:
                 ret['result'] = False
                 ret['comment'] += ' Match text "{0}" was not found.'.format(match)
         elif match_type == 'pcre':
-            if re.search(match, data.get('text', '')):
+            if re.search(str(match), str(data.get('text', ''))):
                 ret['result'] = True
                 ret['comment'] += ' Match pattern "{0}" was found.'.format(match)
             else:
                 ret['result'] = False
                 ret['comment'] += ' Match pattern "{0}" was not found.'.format(match)
 
     if status is not None:
-        if data.get('status', '') == status:
-            ret['comment'] += 'Status {0} was found, as specified.'.format(status)
-            if ret['result'] is None:
-                ret['result'] = True
-        else:
-            ret['comment'] += 'Status {0} was not found, as specified.'.format(status)
-            ret['result'] = False
+        if status_type == 'string':
+            if data.get('status', '') == str(status):
```

Comment:
Should this be `str(data.get('status', '')) == str(status)`? Assuming so, we should have a test for this case.